server/src/coverage/KcovCoverageTool.cpp

```diff
@@ -0,0 +1,131 @@
+//
+// Created by andrey on 05.06.2022.
+//
+
+#include "KcovCoverageTool.h"
+
+#include "Coverage.h"
+#include "Paths.h"
+#include "TimeExecStatistics.h"
+#include "environment/EnvironmentPaths.h"
+#include "exceptions/CoverageGenerationException.h"
+#include "utils/ArgumentsUtils.h"
+#include "utils/CollectionUtils.h"
+#include "utils/FileSystemUtils.h"
+#include "utils/JsonUtils.h"
+#include "utils/MakefileUtils.h"
+#include "utils/StringUtils.h"
+#include "utils/path/FileSystemPath.h"
+#include "pugixml.hpp"
+#include <regex>
+
+#include "loguru.h"
+
+using Coverage::CoverageMap;
+using Coverage::FileCoverage;
+
+
+KcovCoverageTool::KcovCoverageTool(utbot::ProjectContext projectContext,
+                                   ProgressWriter const *progressWriter)
+        : CoverageTool(progressWriter), projectContext(projectContext) {
+}
+
+std::vector<BuildRunCommand>
+KcovCoverageTool::getBuildRunCommands(const std::vector<UnitTest> &testsToLaunch, bool withCoverage) {
+    ExecUtils::throwIfCancelled();
+
+    std::vector<BuildRunCommand> result;
+    ExecUtils::doWorkWithProgress(
+            testsToLaunch, progressWriter, "Collecting build and run commands",
+            [&](UnitTest const &testToLaunch) {
+                auto makefile = Paths::getMakefilePathFromSourceFilePath(
+                        projectContext,
+                        Paths::testPathToSourcePath(projectContext, testToLaunch.testFilePath));
+                auto gtestFlags = getTestFilter(testToLaunch);
+                auto buildCommand =
+                        MakefileUtils::makefileCommand(projectContext, makefile, "build", gtestFlags);
+                auto runCommand =
+                        MakefileUtils::makefileCommand(projectContext, makefile, "run", gtestFlags);
+                result.push_back({testToLaunch, buildCommand, runCommand});
+            });
+    return result;
+}
+
+std::vector<ShellExecTask>
+KcovCoverageTool::getCoverageCommands(const std::vector<UnitTest> &testsToLaunch) {
+    return {};
+}
+
+
+static void addLine(uint32_t lineNumber, bool covered, FileCoverage &fileCoverage) {
+    assert(lineNumber > 0);
```

Comment:
Rather than calling the `assert` function which causes server to crash, please throw an exception